### ISSUE TITLE
Error in log and stop scrape from exporters

### ISSUE THREAD

**Issue by svesh** [VictoriaMetrics/VictoriaMetrics#971]
**Describe the bug**
After some time we have problem with some random exporters:
```
no free connections available to host
```
Similar issue https://github.com/valyala/fasthttp/issues/718

**To Reproduce**
Victoria Metrics single with 5000 targets and 30s scrape interval

**Version**
```
$ ./victoria-metrics-prod --version
victoria-metrics-20201205-115122-tags-v1.49.0-0-gae1db8fa0
```

**Used command-line flags**
```
/victoria-metrics-prod -retentionPeriod=1 -selfScrapeInterval=30s -storageDataPath=/config/data -promscrape.config=/config/prometheus.yml -search.maxUniqueTimeseries=100000 -memory.allowedPercent=15 -search.latencyOffset=30s -maxConcurrentInserts=2000 -search.maxConcurrentRequests=256 -maxLabelsPerTimeseries=100
```

**Comment by f41gh7** [comment]
Can you try flag `-promscrape.streamParse` as temporary workaround, does it help? StreamParse uses default http client without MaxHosts limit, it may also reduce memory usage for scraping huge count of metrics.

**Comment by svesh** [comment]
> Can you try flag `-promscrape.streamParse` as temporary workaround, does it help? StreamParse uses default http client without MaxHosts limit, it may also reduce memory usage for scraping huge count of metrics.

Thank you for recommendation. I applied this flag. We need to wait for some hours to check result.

**Comment by svesh** [comment]
With `-promscrape.streamParse` all work fine. Thank you.

**Comment by valyala** [comment]
This looks like a bug in `fasthttp.HostClient`, which may result in connections leak. Every target is scraped by a dedicated `fasthttp.HostClient`. It can open up to 512 concurrent connection to the target. But under normal conditions it must open only a single connection to the target, since it scrapes it in a single goroutine with `scrape_interval` interval between scrapes. The number of connections can grow to values higher than 1 only on scrape timeouts. The number of connections can grow to big values if scrape interval is too small and all the scrapes are timed out.

**Comment by valyala** [comment]
This issue should be fixed in [v1.97.0](https://docs.victoriametrics.com/CHANGELOG.html), since it substitutes `fasthttp.HostClient` with `net/http.Client` in `vmagent` - see bc7cf4950b5ff1072a33fe58c6bae1184d01068b .
